chain/indexer.go

```diff
@@ -819,6 +836,6 @@ type MessageExecutionProcessor interface {
 type ActorProcessor interface {
 	// ProcessActor processes a set of actors. If error is non-nil then the processor encountered a fatal error.
 	// Any data returned must be accompanied by a processing report.
-	ProcessActors(ctx context.Context, ts *types.TipSet, pts *types.TipSet, actors map[string]types.Actor) (model.Persistable, *visormodel.ProcessingReport, error)
+	ProcessActors(ctx context.Context, ts *types.TipSet, pts *types.TipSet, actors map[string]types.Actor, emsgs []*lens.ExecutedMessage) (model.Persistable, *visormodel.ProcessingReport, error)
```

Comment:
I _almost_ rather just special case the miner actor with its own processor. Does any other actor processor use the `ExecutedMessage` list?

Reply:
No it's the only one. It would be cleaner if we somehow made the messages available through the `lens.Interface`.

The actor extractors themselves now use almost none of the lens interface: just the methods on [ActorStateAPI](https://github.com/filecoin-project/sentinel-visor/blob/iand-missing-block/tasks/actorstate/actorstate.go#L31) which is now just 3 methods with optimization potential plus a way to get a store.

I think there's an opportunity to really simplify this and pass in a type that contains the executed messages and provides access for loading actor state.

Reply:
> It would be cleaner if we somehow made the messages available through the lens.Interface

Agreed. I'd like to rework that interface a bit, or at least reduce the number of implementations (to just lily) as well.

I also suspect the `StateMinerPower` method call to be slow as it loads the power actor every time, although maybe there is already some caching implemented there.